lib/rule-tester/rule-tester.js

```diff
@@ -1096,6 +1108,7 @@ class RuleTester {
                     );
                 } else {
                     assert.strictEqual(result.output, item.output, "Output is incorrect.");
+                    assert.notStrictEqual(item.code, item.output, "Either omit the 'output' property or set it to null if there is no autofix.");
```

Comment:
```suggestion
                    assert.notStrictEqual(item.code, item.output, "Test error object 'output' matches 'code'. If no autofix is expected, then omit the 'output' property or set it to null.");
```

I think this error message should first describe the problem it found, which is that 'output' is the same as 'code' in the test case.